app/javascript/packages/form-steps/use-history-param.spec.tsx

```diff
@@ -144,6 +114,93 @@ describe('useHistoryParam', () => {
     userEvent.clear(input);
     userEvent.type(input, 'one hundred');
 
-    expect(window.location.hash).to.equal('#the%20count=one%20hundred');
+    expect(window.location.hash).to.equal('#one%20hundred');
+  });
+
+  context('with basePath', () => {
+    context('without initial value', () => {
+      beforeEach(() => {
+        const history: string[] = ['/base/'];
+        defineProperty(window, 'location', {
+          value: {
+            get pathname() {
+              return history[history.length - 1];
+            },
+          },
+        });
+
+        sandbox.stub(window.history, 'pushState').callsFake((_data, _unused, url) => {
+          history.push(url as string);
+        });
+        sandbox.stub(window.history, 'back').callsFake(() => {
+          history.pop();
+          window.dispatchEvent(new CustomEvent('popstate'));
+        });
```

Comment:
JSDOM sadly doesn't support any proper page navigation, so we get to roll our own `pushState` implementation!